pkg/homedir/homedir_unix.go

```diff
@@ -133,6 +135,33 @@ func GetConfigHome() (string, error) {
 	return rootlessConfigHomeDir, rootlessConfigHomeDirError
 }
 
+// checkBootId checks if the boot_id of the system has changed since the last time the runtime directory was created.
+func checkBootId(path string) error {
+	systemBootId, err := os.ReadFile("/proc/sys/kernel/random/boot_id")
+	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			// The system does not have a boot_id, so ignore the check
+			return nil
+		}
+		return err
+	}
+
+	bootIdFile := filepath.Join(path, "bootid")
+	bootId, err := os.ReadFile(bootIdFile)
+	if err != nil {
+		// If the file does not exist, create it
+		if errors.Is(err, os.ErrNotExist) {
+			return ioutils.AtomicWriteFile(bootIdFile, systemBootId, 0o600)
+		}
+		return err
+	}
+
+	if string(bootId) != string(systemBootId) {
+		return fmt.Errorf("The boot_id of the system has changed, the runtime directory %q is not valid anymore, please remove it first", path)
```

Comment:
“The boot_id of the system has changed” is an implementation explanation, but it is not directly in terms the user works with (my first instinct might be to search for ways how `boot_id` can be preserved).

Maybe something like “the runtime directory %q has persisted across a reboot”? I can’t think of a short enough explanation.

Absolutely non-blocking: Would we want two versions of this message, one for can’t-happen situations (`/run/user` persisted; just say so) and one for “sad but inevitable situations” (`/var/tmp/storage-run-500`, which we fell back to, was not cleared; give more guidance about what to set up? I have no idea how frequently users run into this.